packages/agents-openai/src/openaiResponsesModel.ts

```diff
@@ -450,12 +470,71 @@ function getInputItems(
           id: item.id!,
           result: item.providerData?.result ?? null,
           status: ImageGenerationStatus.parse(item.status ?? 'failed'),
-          ...item.providerData,
+          ...camelToSnakeCase(item.providerData),
         };
 
         return entry;
       }
 
+      if (
+        item.providerData?.type === 'mcp_list_tools' ||
+        item.name === 'mcp_list_tools'
+      ) {
+        const providerData =
+          item.providerData as ProviderData.HostedMCPListTools;
+        const entry: OpenAI.Responses.ResponseInputItem.McpListTools = {
+          type: 'mcp_list_tools',
+          id: item.id!,
+          tools: camelToSnakeCase(providerData.tools) as any,
+          server_label: providerData.serverLabel,
+          ...camelToSnakeCase(item.providerData),
+        };
+        return entry;
+      } else if (
+        item.providerData?.type === 'mcp_approval_request' ||
+        item.name === 'mcp_approval_request'
+      ) {
+        const providerData =
+          item.providerData as ProviderData.HostedMCPApprovalRequest;
+        const entry: OpenAI.Responses.ResponseInputItem.McpApprovalRequest = {
+          type: 'mcp_approval_request',
+          id: providerData.id ?? item.id!,
```

Comment:
having most of the properties in providerData is easier to understand for maintainers on the agents-core side plus the data size is ignorable, so intentionally duplicating the data